StRoot/StEvent/StGmtHit.h

```diff
@@ -0,0 +1,136 @@
+/**
+ * \class StGmtHit
+ * \brief Holds data for the hit in GMT
+ * 
+ * Data for an individual ``hit'' in GMT, i.e. a 1D cluster (based on StFgtHit).
+ *
+ * \author K.S. Engle, Jan. 2013
+ * \author Richard Witt (witt@usna.edu), Jan. 2013
+ * \author Grigory Nigmatkulov (nigmatkulov@gmail.com), Dec. 2020
+ */
+
+#ifndef StGmtHit_hh
+#define StGmtHit_hh
+
+// C++ headers
+#include <map>
+
+// StEvent headers
+#include "StHit.h"
+#include "StGmtStrip.h"
+
+//________________
+class StGmtHit : public StHit {
+ public:
+  /// Constructor
+  StGmtHit( Int_t key = -1, Int_t module = -1, Float_t adcX = 0, 
+            Float_t adcY = 0, Float_t dadcX = 0, Float_t dadcY = 0,
+	        Float_t localX = 0, Float_t localY = 0,
+	        Float_t localXErr = 10000, Float_t localYErr = 10000,
+	        Float_t sigmaX = 0, Float_t sigmaY = 0,
+	        Float_t sigmaXErr = 10000, Float_t sigmaYErr = 10000);
+  /// Destructor
+  ~StGmtHit();
+
+  /// Print hit information (parameters)
+  void Print(Option_t *option="") const;
+
+  //
+  // Getters
+  //
+
+  /// Unique detector ID
+  virtual StDetectorId detector() const { return kGmtId; }
+
+  /// Key
+  Int_t getKey() const         { return mKey; }
+  /// Module 
+  Int_t getModule() const      { return hardwarePosition() - 1; }
+  /// ADC in X
+  Float_t getAdcX() const      { return mAdcX; }
+  /// ADC error in X
+  Float_t getErrorAdcX() const { return mdAdcX; }
+  /// ADC in Y
+  Float_t getAdcY() const      { return mAdcY; }
+  /// ADC error in Y
+  Float_t getErrorAdcY() const { return mdAdcY; }
+  /// Local X coordinate
+  Float_t getLocalX() const    { return position().x(); }
```

Comment:
The optimizer probably avoids a copy here... but for consistency with the way you write the setters, I recommend using mPosition.x() rather than position().x().  (same for error and y...)

Reply:
Agreed, but that cannot be done here due to the inheritance